### PR TITLE
MaxMind: Feature - PAPP-23253 Add "update data" headers in data for better usability

### PR DESCRIPTION
**Pull request by paulnawat**
Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [X] Testing of all the changes has been performed (for bug fixes / features)
- [X] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [X] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [X] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [X] Verify all checks are passing.
- [X] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [X] Do NOT submit updates to dependencies unless it fixes an issue.
- [X] Limit your pull request to one application and submit multiple pull requests if necessary.
#### Please check the type of change your PR introduces:
- [ ] New App
- [ ] Bugfix
- [X] Feature
- [X] Code style update (formatting, renaming)
- [X] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe): 

## Release Notes (REQUIRED)
- Database information is now included in the result data for better usability. Users can find the last modified data of the database to ensure they are using the expected database version.

## What is the current behavior? (OPTIONAL)
- Users can only see the success message. Users can only know that they are using the latest database.

## What is the new behavior? (OPTIONAL)
- Users can now see the information of the downloaded database. This give more visibility to which database they are using.

## Other information (OPTIONAL)
- This will make it much easier to debug and also remove the install warning.

---
Thanks for contributing!

### CODE REVIEW

File: maxmind.json [hunk 1]
```diff
@@ -338,6 +338,118 @@
                         "failed"
                     ]
                 },
+                {
+                    "data_path": "action_result.data.*.Date",
```
Comment (paulnawat) [inline maxmind.json:342]: Data path here are copied paste from:
image

File: maxmind_connector.py [hunk 2]
```diff
@@ -276,6 +277,8 @@ def _download_db(self, save_path, chunk_size=128):
             for chunk in r.iter_content(chunk_size=chunk_size):
                 fd.write(chunk)
 
+        return headers
```
Comment (paulnawat) [inline maxmind_connector.py:280]: previously, we just succeed without returning anything.

File: maxmind_connector.py [hunk 3]
```diff
@@ -249,12 +249,13 @@ def _handle_update_db(self, param):
...
 
         try:
-            status, msg = self._download_and_replace_db()
+            status, msg, response_headers = self._download_and_replace_db()
             action_result.set_status(status, msg)
         except Exception as e:
             error_msg = 'Error in downloading or replacing database.'
             action_result.set_status(phantom.APP_ERROR, error_msg, e)
 
+        action_result.add_data(dict(response_headers))
```
Comment (paulnawat) [inline maxmind_connector.py:258]: Simply include the headers in the final result. 
This is what the final result looks like:
image

File: readme.md [hunk 4]
```diff
@@ -1,12 +1,12 @@
 [comment]: # "Auto-generated SOAR connector documentation"
 # MaxMind
 
-Publisher: Splunk
```
Comment (paulnawat) [inline readme.md:4]: This file is auto updated from running `pre-commit run --all-files`

File: maxmind.json (hunk 1 above)
Comment (paulnawat) [reply maxmind.json:342]: This should be more clear in the README as well:
image

File: maxmind_connector.py [hunk 5]
```diff
@@ -301,11 +304,11 @@ def _replace_db(self, tar_file_path):
     def _download_and_replace_db(self):
         self.debug_print('Downloading database.')
 
-        self._download_db(MMDB_ZIP_FILE_PATH)
+        response_headers = self._download_db(MMDB_ZIP_FILE_PATH)
         self._replace_db(MMDB_ZIP_FILE_PATH)
 
         self.debug_print('Successfully updated database.')
-        return phantom.APP_SUCCESS, 'Successfully updated database.'
+        return (phantom.APP_SUCCESS, 'Successfully updated database.', response_headers)
```
Comment (rgil-splunk) [inline maxmind_connector.py:311]: you don't need parentheses to return multiple items :)

**Review by rgil-splunk — APPROVED** [review]

**Pull request merged**